src/connector/src/source/cdc/external/postgres.rs

```diff
@@ -256,6 +257,126 @@ impl PostgresExternalTableReader {
+}
+
+pub fn pg_type_to_rw_type(pg_type: &PgType) -> ConnectorResult<DataType> {
+    let data_type = match *pg_type {
+        PgType::BOOL => DataType::Boolean,
+        PgType::BIT => DataType::Boolean,
+        PgType::INT2 => DataType::Int16,
+        PgType::INT4 => DataType::Int32,
+        PgType::INT8 => DataType::Int64,
+        PgType::FLOAT4 => DataType::Float32,
+        PgType::FLOAT8 => DataType::Float64,
+        PgType::NUMERIC | PgType::MONEY => DataType::Decimal,
+        PgType::DATE => DataType::Date,
+        PgType::TIME => DataType::Time,
+        PgType::TIMETZ => DataType::Time,
+        PgType::POINT => DataType::Struct(risingwave_common::types::StructType::new(vec![
+            ("x", DataType::Float32),
+            ("y", DataType::Float32),
+        ])),
+        PgType::TIMESTAMP => DataType::Timestamp,
+        PgType::TIMESTAMPTZ => DataType::Timestamptz,
+        PgType::INTERVAL => DataType::Interval,
+        PgType::VARCHAR | PgType::TEXT | PgType::BPCHAR | PgType::UUID => DataType::Varchar,
+        PgType::BYTEA => DataType::Bytea,
+        PgType::JSON | PgType::JSONB => DataType::Jsonb,
+        // Array types
+        PgType::BOOL_ARRAY => DataType::List(Box::new(DataType::Boolean)),
+        PgType::BIT_ARRAY => DataType::List(Box::new(DataType::Boolean)),
+        PgType::INT2_ARRAY => DataType::List(Box::new(DataType::Int16)),
+        PgType::INT4_ARRAY => DataType::List(Box::new(DataType::Int32)),
+        PgType::INT8_ARRAY => DataType::List(Box::new(DataType::Int64)),
+        PgType::FLOAT4_ARRAY => DataType::List(Box::new(DataType::Float32)),
+        PgType::FLOAT8_ARRAY => DataType::List(Box::new(DataType::Float64)),
+        PgType::NUMERIC_ARRAY => DataType::List(Box::new(DataType::Decimal)),
+        PgType::VARCHAR_ARRAY => DataType::List(Box::new(DataType::Varchar)),
+        PgType::TEXT_ARRAY => DataType::List(Box::new(DataType::Varchar)),
+        PgType::BYTEA_ARRAY => DataType::List(Box::new(DataType::Bytea)),
+        PgType::DATE_ARRAY => DataType::List(Box::new(DataType::Date)),
+        PgType::TIME_ARRAY => DataType::List(Box::new(DataType::Time)),
+        PgType::TIMESTAMP_ARRAY => DataType::List(Box::new(DataType::Timestamp)),
+        PgType::TIMESTAMPTZ_ARRAY => DataType::List(Box::new(DataType::Timestamptz)),
+        PgType::INTERVAL_ARRAY => DataType::List(Box::new(DataType::Interval)),
+        PgType::JSON_ARRAY => DataType::List(Box::new(DataType::Jsonb)),
+        PgType::JSONB_ARRAY => DataType::List(Box::new(DataType::Jsonb)),
+        PgType::UUID_ARRAY => DataType::List(Box::new(DataType::Varchar)),
+        PgType::OID => DataType::Int64,
+        PgType::OID_ARRAY => DataType::List(Box::new(DataType::Int64)),
+        PgType::POINT_ARRAY => DataType::List(Box::new(DataType::Struct(
+            risingwave_common::types::StructType::new(vec![
+                ("x", DataType::Float32),
+                ("y", DataType::Float32),
+            ]),
+        ))),
```

Comment:
is it possible to expand with macro?

Reply:
I think using macros here is a bit of an over-optimization, and it doesn't seem to save much code.🤔